Assets/HoloToolkit/Input/README.md

```diff
@@ -3,50 +3,649 @@
+
+The `InputManager` forwards the various input sources events to the appropriate `GameObject`, using the following order:
+
+1. Registered Global Listeners
+2. The registered modal input handlers, in LIFO (Last-In First-Out) order of registration
+3. The currently focused `GameObject`
+4. The fallback input handlers, in LIFO (Last-In First-Out) order of registration
+
+`GameObject`s that want to consume input events can implement one or more input interfaces.  These interface handlers follow a specific hierarchy:
+
+![alt text](/External/ReadMeImages/input_system_handler_hierarchy.PNG)
+
+### Generic Cross Platform Interfaces
+
+- `ISourceStateHandler` for all source detected/source lost events.
+- `IFocusable` for focus enter and exit. The focus can be triggered by the user's gaze or any other gaze source.
+- `IInputHandler` for source up, down, and pressed events.
+    - Similar to [Unity’s `Input` class](https://docs.unity3d.com/ScriptReference/Input.html) `GetButtonDown`, `GetButtonUp`, and `GetButton` methods.
+    - Only Pointers will raise UI events for Source Up and Source Down.
+- `IInputClickHandler` for source “clicked”.  The Default Pointer Sources that raise the click handler include:
+    - Mouse Clicks
+    - When a user says “Select”
+    - Hand Taps
+    - Clicker Presses
+    - Motion Controller Trigger Presses
+- `IXboxControllerHandler` for Cross Platform Xbox One Controller events.
+
+### Windows Specific Interfaces
 
-Each input source (hands, gestures, others) implement a **IInputSource** interface. The interface defines various events that the input sources can trigger. The input sources register themselves with the InputManager, whose role it is to forward input to the appropriate game objects. Input sources can be dynamically enabled / disabled as necessary, and new input sources can be created to support different input devices.
+- `ISpeechHandler` for voice commands.
+- `IDictationHandler` for speech to text dictation.
 
-Game objects that want to consume input events can implement one or many **input interfaces**, such as:
+### UWP Specific Interfaces
 
-- **IFocusable** for focus enter and exit. The focus can be triggered by the user's gaze or any other gaze source.
-- **IHoldHandle** for the Windows hold gesture.
-- **IInputHandler** for source up and down. The source can be a hand that tapped, a clicker that was pressed, etc.
-- **IInputClickHandler** for source clicked. The source can be a hand that tapped, a clicker that was pressed, etc.
-- **IManipulationHandler** for the Windows manipulation gesture.
-- **INavigationHandler** for the Windows navigation gesture.
-- **ISourceStateHandler** for the source detected and source lost events.
-- **ISpeechHandler** for voice commands.
-- **IDictationHandler** for speech to text dictation.
-- **IGamePadHandler** for generic gamepad events.
-- **IXboxControllerHandler** for Xbox One Controller events.
+- `IHoldHandler` for the Windows hold gesture.
```

Comment:
Can you explain here, what is the "Hold" gesture? Or put a link to a description? And maybe a reference to where I can find related informations, like what to do next, when I have detected the Hold gesture. Is this related to the `IManipulationHandler`?

Reply:
The link to include would probably be https://developer.microsoft.com/en-us/windows/mixed-reality/gestures#gesture_recognizers

The "Hold" gesture is the "Tap and hold" composite gesture defined on that page. It's not related to the manipulation handler: manipulation is its own gesture.